### PR TITLE
fix: login to CF before deploying mtar

### PR DESCRIPTION
**Pull request by rayakoren**
Before submitting a pull request, please ensure the following:

## Description

Please explain the changes you made.

Add a link to the design (if applicable).

### Checklist

- [ ] Code compiles correctly
- [ ] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [ ] Linting run locally successfully
- [ ] All tests pass
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: src/commands/mtarDeployCommand.ts [hunk 1]
```diff
@@ -78,15 +78,35 @@ export class MtarDeployCommand {
...
+    // check if the user is logged in to CF
+    const isLoggedIin = await Utils.isLoggedInToCfWithProgress();
+    if (isLoggedIin) {
+      return true;
+    }
+
+    // connect the user to CF
+    const weakLogin = true; //if partially logged (example: only org but not space) ask the user only for the missing "questions"
+    const allCommands = await commands.getCommands(true);
+    if (includes(allCommands, CF_LOGIN_COMMAND)) {
```
Comment (alex-gilin) [inline src/commands/mtarDeployCommand.ts:97]: there is a 'cf.login.weak' command also

File: src/utils/utils.ts [hunk 2]
```diff
@@ -145,6 +147,39 @@ export class Utils {
...
+
+  public static async isLoggedInToCf(): Promise<boolean> {
+    try {
+      const target: ITarget = await this.getTarget();
+      // user is connected to CF
+      if (
+        !isEmpty(target.user) &&
+        !isEmpty(target.org) &&
+        !isEmpty(target.space) &&
+        !isEmpty(target["api endpoint"])
```
Comment (alex-gilin) [inline src/utils/utils.ts:162]: redundant. if user attribute is fulfilled the endpoint must exist

File: src/utils/utils.ts [hunk 3]
```diff
@@ -145,6 +147,39 @@ export class Utils {
...
+      // User is not logged in to Cloud Foundry
+    }
+    return false;
+  }
+
+  public static async isLoggedInToCfWithProgress(): Promise<boolean> {
+    return window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: "Checking your connection to Cloud Foundry...",
```
Comment (alex-gilin) [inline src/utils/utils.ts:176]: KM (Paola)? candidate to extract in static messages ?

**Review by alex-gilin — CHANGES_REQUESTED** [review]
you need to bump a package version, isn't ?

File: src/utils/utils.ts (hunk 2 above)
Comment (rayakoren) [reply src/utils/utils.ts]: Done

File: src/commands/mtarDeployCommand.ts (hunk 1 above)
Comment (rayakoren) [reply src/commands/mtarDeployCommand.ts:96]: Done

**Comment by rayakoren** [comment]
> you need to bump a package version, isn't ?

The bump is done automatically when we create tag...

File: src/utils/utils.ts (hunk 3 above)
Comment (rayakoren) [reply src/utils/utils.ts:175]: asked Paola/Shira to review

Comment (rayakoren) [reply src/utils/utils.ts:175]: Paola approved

**Review by alex-gilin — APPROVED** [review]

**Review by michawai — APPROVED** [review]

**Pull request merged**